library/src/containers/Servers/ServerSecurity.tsx

```diff
@@ -8,82 +8,165 @@ import { ServerHelpers } from '../../helpers';
 
 interface Props {
   serverRequirements: ServerSecurityRequirement[];
+  protocol: string;
 }
 
 export const ServerSecurity: React.FunctionComponent<Props> = ({
   serverRequirements = [],
+  protocol = '',
 }) => {
   const asyncapi = useSpec();
   const securitySchemes =
     asyncapi.hasComponents() && asyncapi.components().securitySchemes();
 
+  let renderedServerSecurities;
   if (
+    !serverRequirements ||
     !serverRequirements.length ||
     !securitySchemes ||
     !Object.keys(securitySchemes).length
   ) {
-    return null;
-  }
-
-  const serverSecurities: React.ReactNodeArray = serverRequirements
-    .map(requirement => {
-      const def: SecurityScheme =
-        securitySchemes[Object.keys(requirement.json())[0]];
-
-      if (!def) {
-        return null;
-      }
-      return <ServerSecurityItem securitySchema={def} key={def.type()} />;
-    })
-    .filter(Boolean);
-
-  if (!serverSecurities.length) {
-    return null;
-  }
+    if (protocol === 'kafka' || protocol === 'kafka-secure') {
+      renderedServerSecurities = (
+        <ServerSecurityItem protocol={protocol} securitySchema={null} key="" />
```

Comment:
If you wanna render single element, you don't need `key=""` prop:

```suggestion
        <ServerSecurityItem protocol={protocol} securitySchema={null} />
```